sparrow_datums/keypoints/keypoints.py

```diff
@@ -0,0 +1,172 @@
+from typing import TypeVar
+
+import numpy as np
+
+from ..chunk import Chunk
+from ..exceptions import ValidationError
+from ..types import FloatArray, PType
+
+T = TypeVar("T", bound="Keypoints")
+
+
+class Keypoints(Chunk):
+    """Dense data arrays for keypoints.
+
+    Parameters
+    ----------
+    Chunk : Chunk
+        Base class for dense data arrays with metadata.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from sparrow_datums import Keypoints, PType
+    >>> Keypoints(np.ones(2))
+    Keypoints([1., 1.])
+    """
+
+    def validate(self) -> None:
+        """Check validity of keypoints array."""
+        if not self.shape or self.shape[-1] != 2:
+            raise ValidationError("Keypoint arrays must have size-2 dimensions")
+
+    @property  # Overriding
+    def scale(self) -> FloatArray:
+        """Scaling array."""
+        if self._scale is None or len(self._scale) == 4:
```

Comment:
Can you think of a case where `len(self._scale)` would be 4 for a keypoint chunk?

Reply:
yeah the reason for putting len(self._scale) == 4 constrain is that the keypoint class inherits from the chunk class. I am trying to override the scale function from the parent class (chunk) which sets the scale to 4 dimensions. Perhaps I am missing something here?

Reply:
Oh yep good point. The way `scale` is defined on the parent is box-specific. Would you be willing to move that property to the `Boxes` base class instead of `Chunk`? That could break some tests that will need to be fixed, but that's really where it belongs.

Reply:
sounds good